isis/src/system/apps/isisdataeval/isisdataeval.xml

```diff
@@ -0,0 +1,867 @@
+error,/opt/isisdatafull/tgo/kernels/dsk/em16_tgo_sc_sa,/opt/isisdatafull/tgo/kernels/dsk/em16_tgo_sc_sa+z_v00.obj,/opt/isisdatafull/tgo/kernels/dsk/em16_tgo_sc_sa,/opt/isisdatafull/tgo/kernels/dsk/em16_tgo_sc_sa+z_v00.obj,openfailed
+error,/opt/isisdatafull/tgo/kernels/dsk/em16_tgo_sc_sa,/opt/isisdatafull/tgo/kernels/dsk/em16_tgo_sc_sa+z_v00.png,/opt/isisdatafull/tgo/kernels/dsk/em16_tgo_sc_sa,/opt/isisdatafull/tgo/kernels/dsk/em16_tgo_sc_sa+z_v00.png,openfailed
+error,/opt/isisdatafull/voyager1/kernels/spk/voyager_1.ST,/opt/isisdatafull/voyager1/kernels/spk/voyager_1.ST+1991_a54418u.merged.bsp,/opt/isisdatafull/voyager1/kernels/spk/voyager_1.ST,/opt/isisdatafull/voyager1/kernels/spk/voyager_1.ST+1991_a54418u.merged.bsp,openfailed
+error,/opt/isisdatafull/voyager1/kernels/spk/voyager_2.ST,/opt/isisdatafull/voyager1/kernels/spk/voyager_2.ST+1992_m05208u.merged.bsp,/opt/isisdatafull/voyager1/kernels/spk/voyager_2.ST,/opt/isisdatafull/voyager1/kernels/spk/voyager_2.ST+1992_m05208u.merged.bsp,openfailed
+error,/opt/isisdatafull/voyager2/kernels/spk/voyager_1.ST,/opt/isisdatafull/voyager2/kernels/spk/voyager_1.ST+1991_a54418u.merged.bsp,/opt/isisdatafull/voyager2/kernels/spk/voyager_1.ST,/opt/isisdatafull/voyager2/kernels/spk/voyager_1.ST+1991_a54418u.merged.bsp,openfailed
+error,/opt/isisdatafull/voyager2/kernels/spk/voyager_2.ST,/opt/isisdatafull/voyager2/kernels/spk/voyager_2.ST+1992_m05208u.merged.bsp,/opt/isisdatafull/voyager2/kernels/spk/voyager_2.ST,/opt/isisdatafull/voyager2/kernels/spk/voyager_2.ST+1992_m05208u.merged.bsp,openfailed
+
+Group = Results
+  ISISDATA            = /opt/isisdatafull
+  DATADIR             = /opt/isisdatafull
+  EmptyKernelDBs      = 11
+  MissingKernelDBs    = 23
+  SymlinkKernelFiles  = 0
+  ExternalKernelFiles = 11
+  TotalDBConfigFiles  = 10 &lt;conf&gt;
+  TotalKernelDBFiles  = 219 &lt;db&gt;
+  TotalDirectories    = 565
+  TotalDataFiles      = 101452
+  TotalInstallSize    = 1947030252490 &lt;bytes&gt;
+  TotalVolumeSize     = 1813.3132276032 &lt;GB&gt;
+  InventorySymLinks   = 0
+  HashBufferSize      = 268435456 &lt;bytes&gt;
+  TotalVolumeHash     = 6cd919230b88ba0fa68d71d9942f3894 &lt;md5&gt;
+End_Group
+</PRE>
+          The files labeled <em>missing</em> are kernel files contained in
+          either a kernels.????.db or kernels.????.conf ISIS kernel database
+          configuration file. Files with <em>empty</em> status have no
+          content or do not specify any kernels. This may be needed to
+          satisfy spiceinit requirements. A status of <em>external</em>
+          indicate the file <em>as specified</em> in the db/conf file
+          does not exist in the ISISDATA inventory. For the occurances
+          in the above results indicate an extra '/' in the file name.
```

Comment:
just to be clear, external means that it is a file NOT mentioned in any .db and therefore essential a dead kernel to ISIS?

Reply:
No. _external_ file status as determined in **isisdateaveal** has several meanings that may include:

- A file that is referred to in a kernel DB/conf file, but is not found in the full ISISDATA file inventory (it can still exist but the full path to the file must contain the expanded/specified $ISISDATA path or its deemed external). These situations are common when ISISDATA == DATADIR.
- A misspelled kernel file specification that contains an extra / in its file name (as seen in Voyager SPK kernel configs in #5110). The file is present and can be read in ISIS with the file specification, but it's still identified as an issue nonetheless.
- When you choose DATADIR as a mission directory in $ISISDATA, e.g., DATADIR=$ISISDATA/osirisrex, any file specified in a kernel DB or conf file that is not contained within the $ISISDATA/osirisrex directory hierarchy (i.e., _inventory_), is considered external. This identifies ISISDATA dependencies outside the mission directory even if the file is in $ISISDATA but not in DATADIR - a quite useful feature. And these cases may not indicate an error, but it is good to know from a management perspective (if your ISIS mission kernel spec has outside dependencies).

Note its difficult to make assumptions in **isisdataeval** regarding a dead file because all files in ISISDATA are considered inventory, including calibration files, which are typically not part of the kernel infrastructure.

Reply:
This seems kinda broad, does it make sense to make these different classifications? It seems useful to flag files that match case 2 and 3 specifically rather than being pooled together.

Reply:
What classification do you suggest?

Case 2 is difficult to identify specifically as a misspelled file in the config file and may have different OS behaviors. It is determined by string comparisons of the environment variable expansion and the result of absolute path by OS resources via [Qt::QFileInfo](https://doc.qt.io/qt-6/qfileinfo.html#absoluteFilePath) which seems to clean up the path.

I can add more documentation based upon this narrative if that will be helpful.

Reply:
I think for now the extra text in the XML docs that breaks down these cases is good for now.